CONTRIBUTING.rst

```diff
@@ -264,40 +254,48 @@ leverage multiple processes. Example usage:
 
    python -m pytest -n <NUMCORE> --cov pyvista
 
-Run all code examples in the docstrings. Be sure to export the
-``PYVISTA_OFF_SCREEN`` environmental variable to enable off-screen
-plotting.
+Documentation Testing
+~~~~~~~~~~~~~~~~~~~~~
+Run all code examples in the docstrings with:
 
 .. code:: bash
 
-   export PYVISTA_OFF_SCREEN=True
    python -m pytest -v --doctest-modules pyvista
 
-Run documentation testing by running:
+Build the documentation with:
 
 .. code:: bash
 
-   make
+   make -C doc html
+
+The finished documentation can be found in the ``doc/_build/html``
+directory.
 
-If you are running windows and ``make`` is unavailable, then run:
+Style Checking
+~~~~~~~~~~~~~~
+PyVista follows PEP8 standard as outlined in the `Coding Style section
+<#coding-style>`_ and implements style checking using `pre-commit`_.
 
-.. code:: bash
+To ensure your code meets minimum code styling standards, run::
 
-   pydocstyle pyvista
+  pip install pre-commit
+  pre-commit run --all-files
 
-   codespell pyvista/ examples/ tests/ -S "*.pyc,*.txt,*.gif,*.png,*.jpg,*.ply,*.vtk,*.vti,*.js,*.html,*.doctree,*.ttf,*.woff,*.woff2,*.eot,*.mp4,*.inv,*.pickle,*.ipynb,flycheck*" -I "ignore_words.txt"
+You can also install this as a pre-commit hook by running::
 
-And finally, test the documentation examples:
+  pre-commit install
 
-.. code:: bash
+This way, it's not possible for you to push code that fails the style
+checks. For example, each commit automatically checks that you meet the style
+requirements::
 
-   cd doc
-   make clean
-   make doctest
-   make html -b linkcheck
+  $ pre-commit install
+  $ git commit -m "added my cool feature"
+  black....................................................................Passed
+  isort....................................................................Passed
+  flake8...................................................................Passed
+  codespell................................................................Passed
 
```

Comment:
```suggestion

The actual installation of the environment happens before the first commit 
following ``pre-commit install``. This will take a bit longer, but subsequent
commits will only trigger the actual style checks.
```